src/wayland/view-backend-wayland.cpp

```diff
@@ -181,29 +181,27 @@ ViewBackend::ViewBackend(struct wpe_view_backend* backend)
 
     m_surface = wl_compositor_create_surface(m_display.interfaces().compositor);
 
+    // In case that more than one protocol is available pick the first that matches.
     if (m_display.interfaces().xdg) {
         m_xdgSurface = xdg_shell_get_xdg_surface(m_display.interfaces().xdg, m_surface);
         xdg_surface_add_listener(m_xdgSurface, &g_xdgSurfaceListener, &m_resizingData);
         xdg_surface_set_title(m_xdgSurface, "WPE");
-    }
-
-    if (m_display.interfaces().xdg_v6) {
+    } else if (m_display.interfaces().xdg_v6) {
         m_xdg6Surface = zxdg_shell_v6_get_xdg_surface(m_display.interfaces().xdg_v6, m_surface);
         zxdg_surface_v6_add_listener(m_xdg6Surface, &g_xdg6SurfaceListener, nullptr);
-
         m_toplevelSurface = zxdg_surface_v6_get_toplevel(m_xdg6Surface);
         if (m_toplevelSurface) {
             zxdg_toplevel_v6_add_listener(m_toplevelSurface, &g_toplevelSurfaceListener, &m_resizingData);
             zxdg_toplevel_v6_set_title(m_toplevelSurface, "WPE");
             wl_surface_commit(m_surface);
         }
-    }
-
-    if (m_display.interfaces().ivi_application) {
+    } else if (m_display.interfaces().ivi_application) {
```

Comment:
Again, I am not very sure about the order of choices in the protocol being used, and I wonder whether the `ivi_application` should maybe even be tried first, as it is the more specific of the protocols.

Reply:
I think this is a pretty minor point, but I agree that the order should indeed probably be IVI -> xdgv6 -> xdg.

Reply:
P.S. Garnacho is the person to ask if you want an explanation about the need for the xdgv6 monstrosity.